src/core/stdc/stdlib.d

```diff
@@ -159,6 +159,17 @@ else
 ///
 void*   malloc(size_t size) pure;
 ///
+void*   pureMalloc(size_t size) pure
+{
+    const errno = pureGetErrno();
+    typeof(return) ret = malloc(size);
+    cast(void)pureSetErrno(errno);
+    return ret;
+}
+extern (C) pure pragma(mangle, "getErrno") int pureGetErrno(); // for internal use
+extern (C) pure pragma(mangle, "setErrno") int pureSetErrno(int); // for internal use
```

Comment:
Well just saying that https://github.com/ldc-developers/ldc/issues/1020 doesn't apply to these `extern (C)` fwd declarations. While declaring them with their real name should work too, I think the explicit `pure` in the function name makes a lot of sense in this case.